Automattic/wp-calypso
# Live Preview: Add the upgrade notice for Premium and Woo themes

Related to https://github.com/Automattic/wp-calypso/issues/79223, https://github.com/Automattic/wp-calypso/pull/82629

## Proposed Changes

This PR adds the upgrade-needed notice on the Site Editor canvas when live-previewing Premium and WooCommerce themes. 

**The latest copy:**
<img width="1426" alt="Screenshot 2023-11-28 at 16 21 17" src="https://github.com/Automattic/wp-calypso/assets/5287479/24f23e4a-cb40-4605-902c-a3ad625430e3">
<img width="1432" alt="Screenshot 2023-11-28 at 16 20 12" src="https://github.com/Automattic/wp-calypso/assets/5287479/1389bd8f-525c-4829-bcfb-6947e2cd344d">



**The flow:**

https://github.com/Automattic/wp-calypso/assets/5287479/63985799-0f78-4472-a5a6-97576edbb04a



I'll create a follow-up PR for the following (See https://github.com/Automattic/wp-calypso/issues/79223 for more details).

- [x] the notice on the sidebar https://github.com/Automattic/wp-calypso/pull/84676
- [ ] the modal when clicking the Activate button

## Testing Instructions



- Run `install-plugin.sh wpcom-block-editor add/live-preview-upgrade-notice` on your sandbox.
- Sandbox widgets.wp.com and your site.
- **Upgrade Notice for Premium themes**
	- Prepare a site with the Personal plan or lower plan. 
	- Access `https://<your-site>/wp-admin/site-editor.php?wp_theme_preview=<premium-theme>` to preview a premium theme (e.g., `premium/outland`).
	- Try to edit the homepage and see the upgrade notice.
	- Click `Upgrade now`.
	- Verify the **Premium** plan is in the cart and complete checkout.
	- Verify it redirects to the Site Editor.
	- Verify you don't see the upgrade notice and the plan is upgraded.
- **Upgrade Notice for WooCommerce themes**
	- Prepare a site with the Premium plan or lower plan. 
	- Access `https://<your-site>/wp-admin/site-editor.php?wp_theme_preview=<woo-theme>` to preview a woo commerce theme (e.g. `premium/tsubaki`).
	- Try to edit the homepage and see the upgrade notice.
	- Click `Upgrade now`.
	- Verify the **Business** plan is in the cart and complete checkout.
	- Verify it redirects to the Site Editor.
	- Verify you don't see the upgrade notice and the plan is upgraded.
- **No Upgrade Notices**
	- Prepare a site with the Business plan or higher plan.
	- Access `https://<your-site>/wp-admin/site-editor.php?wp_theme_preview=<premium-theme>` to preview a premium theme (e.g., `premium/outland`). 
		- If your site is on Atomic, you need to install the theme before previewing it. 
	- Verify you don't see the upgrade notice. 

## Pre-merge Checklist



	- I didn't write tests, as most logic is from existing functions and I don’t think it’s a good idea to make it more complicated. I extracted functions and wrote types as much as possible.
